frontend/src/metabase/metadata/pages/DataModel/components/PreviewSection/TablePreview.tsx

```diff
@@ -107,7 +131,10 @@ function getPreviewQuery(
       "source-table": tableId,
       filter,
       fields: [fieldRef],
-      limit: 50, // fetch more rows to increase probability of getting at least 5 unique values
+      // fetch more rows to increase probability of getting at least 5 unique values
+      limit: 50,
+      // order by PKs when possible to prevent SQL returning non-deterministically ordered values
```

Comment:
Try this:

1. New > Question > Orders > Visualize

First few values in "Product ID" column are different.

2. New > Question > Orders > Disable all fields in data step except "Product ID" > Visualize

All first 50+ values in "Product ID" column are the same